languages/java/exercises/concept/strings/.docs/after.md

```diff
@@ -0,0 +1,48 @@
+The key thing to remember about Java strings is that they are immutable objects representing text as a sequence of Unicode characters (letters, digits, punctuation, etc.). Double quotes are used to define a `String` instance:
+
+```java
+String fruit = "Apple";
+```
+
+Manipulating a string can be done using method of class [`String`][string-class]. As string values can never change after having been defined, all string manipulation methods will return a new string.
+
+A string is delimited by double quote (`"`) characters. Some special characters need escaping using the backslash (`\`) character.
+Characters to be escaped in Java:
+
+- `"`
+- `\`
+
+```java
+String escaped = "c:\\test.txt";
+// => c:\test.txt
+```
+
+Finally, there are many ways to concatenate a string. The simplest one is by using the `+` operator.
+Other possibilities are:
+
+- use [`StringBuilder` class][string-builder]
+- use [`String.concat` method][string-concat]
+
+```java
+String name = "Jane";
+"Hello " + name + "!";
+// => "Hello Jane!"
+
+new StringBuilder().append("Hello ").append("Jane").append("!").toString();
```

Comment:
Are the `StringBuilder` and `concat` approaches used often? I suspect the `+` and `String.format` approach are the most common ones, right? If so, I would suggest mentioning those first. You could even consider just referring to the `StringBuilder` or `String.concat` methods without having any code examples. For the C# track, I've omitted the use of the `StringBuilder` at this point, as I want to introduce it in a separate exercise. It is fine to mention/link here of course, but just wanted to give you my reasoning.

Reply:
I think `+` is for sure the approach more used. `StringBuilder` is the best practice when you have to concatenate a lot of strings: it permits to avoid to clutter the memory with a lot temp strings.
I'll try to describe this concept better in `after.md`

Reply:
Great! Note that I've approved this PR, which means that you can merge it as is. You can then do a follow-up PR to fix an open issues.